resources/tests.mjs

```diff
@@ -1113,6 +1113,90 @@ Suites.push({
+    async prepare(page) {
+        (await page.waitForElement("#content-iframe")).focus();
+    },
+    tests: [
+        new BenchmarkTestStep("LoadChatAndExpandRecipes", (page) => {
+            const iframeElement = page.querySelector("#content-iframe", [], true);
+            const resumePreviousChatBtn = iframeElement.querySelectorInShadowRoot("#resume-previous-chat-btn", ["cooking-app", "chat-window"]);
+            resumePreviousChatBtn.click();
+            page.layout();
+
+            // Navigate through the restaurants
+            const nextRestaurantBtn = iframeElement.querySelectorInShadowRoot("#next-restaurant-btn", ["cooking-app", "information-window"]);
+            const restaurantCards = iframeElement.querySelectorAllInShadowRoot("restaurant-card", ["cooking-app", "information-window"]);
+            const numOfRestaurantCards = restaurantCards.length - 1; // since 1 is already visible
+            for (let i = 0; i < numOfRestaurantCards; i++) {
+                nextRestaurantBtn.click();
+                page.layout();
+            }
+
+            // Expand recipes
+            const showMoreBtn = iframeElement.querySelectorAllInShadowRoot(".show-more-btn", ["cooking-app", "main-content", "recipe-grid"]);
+            for (const btn of showMoreBtn) {
+                btn.click();
+                page.layout();
+            }
+        }),
+        new BenchmarkTestStep("ReduceWidthIn5Steps", (page) => {
+            const iframeElement = page.querySelector("#content-iframe");
+            const widths = [768, 704, 640, 560, 480];
+
+            widths.forEach((width) => {
+                iframeElement.setWidth(`${width}px`);
+                page.layout();
+            });
+        }),
+        new BenchmarkTestStep("ScrollToChatAndSendMessages", (page) => {
+            const iframeElement = page.querySelector("#content-iframe", [], true);
+
+            // Navigate through the carousel items
+            const nextItemBtn = iframeElement.querySelectorInShadowRoot("#next-item-carousel-btn", ["cooking-app", "main-content", "recipe-carousel"]);
+            const recipeCarouselItems = iframeElement.querySelectorAllInShadowRoot(".carousel-item", ["cooking-app", "main-content", "recipe-carousel"]);
+            const numOfCarouselItems = recipeCarouselItems.length - 3; // since 3 are already visible
+            for (let i = 0; i < numOfCarouselItems; i++) {
+                nextItemBtn.click();
+                page.layout();
+            }
+
+            // Collapse recipes
+            const showMoreBtn = iframeElement.querySelectorAllInShadowRoot(".show-more-btn", ["cooking-app", "main-content", "recipe-grid"]);
+            for (const btn of showMoreBtn) {
+                btn.click();
+                page.layout();
+            }
+
+            const element = iframeElement.querySelectorInShadowRoot("#chat-window", ["cooking-app", "chat-window"]);
+            element.scrollIntoView();
```

Comment:
We should probably explicitly force the "instant" behavior. If we accidentally get "smooth" it will likely be not measured correctly.